### PR TITLE
fix(animations): always fire inner trigger callbacks even if blocked by parent animations

### LINKED ISSUE #19100: Nested animation don't run and start callback is not fired
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
when 2 nested component with animation trigger defined on each component, only toplevel animation is run and its start animation callback is called. 

## Expected behavior
each callback should be called

## Minimal reproduction of the problem with instructions
https://plnkr.co/edit/Wa7UNmKQkaVMWPNGmoxH?p=preview

## What is the motivation / use case for changing the behavior?
Nested animation are usefull for defined enter animation

## Environment

<pre><code>
Angular version: 4.4.0-RC.0
 
Browser:
- [x] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [x] Firefox version XX
- [x] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX

### PR DESCRIPTION
**Pull request by matsko**
Closes #19100

### CODE REVIEW

File: packages/animations/browser/src/render/transition_animation_engine.ts [hunk 1]
```diff
@@ -1365,6 +1382,15 @@ export class TransitionAnimationPlayer implements AnimationPlayer {
 
   getRealPlayer() { return this._player; }
 
+  syncPlayerEvents(player: AnimationPlayer) {
+    const p = this._player as any;
+    if (p.triggerCallback) {
+      player.onStart(() => p.triggerCallback !('start'));
```
Comment (mhevery) [inline packages/animations/browser/src/render/transition_animation_engine.ts]: remove `!`

**Review by mhevery — APPROVED** [review]

**Comment by tbosch** [comment]
This broke a test in G3.

**Comment by fxck** [comment]
@tbosch any chance this will get in before angular 5 final? It's a requirement for https://github.com/angular/angular/pull/19455 without which nested animations are pretty much broken..

**Review by mhevery — APPROVED** [review]

**Comment by matpag** [comment]
@mhevery This PR seems to have been reverted but it is listed in the current changelog for the 5.1.0-beta1 release. Am i missing something?

**Pull request closed without merging**